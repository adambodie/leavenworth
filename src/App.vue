<template>
  <div id="app">
    <Header />
    <Card />
  </div>
</template>

<script>
import Header from './components/Header'
import Card from './components/Card'

export default {
  name: 'app',
  components: {
    Card,
    Header
  }
}
</script>

<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
}
</style>
