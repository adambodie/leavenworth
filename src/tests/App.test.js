import Vue from 'vue'
import App from '../vue/app.vue'
import { shallow } from 'vue-test-utils'
import axios from 'axios'
import MockAdapter from "axios-mock-adapter";

describe('App.test.js', () => {
	let cmp, vm
	const mock = new MockAdapter(axios);
	
	beforeEach(() => {
		cmp = shallow(App, {})
		cmp.setData({url: "https://api.flickr.com/services/rest/?method=flickr.photosets.getPhotos&api_key=0c3f8d32a28de8434240115b85a28499&photoset_id=72157695078896702&user_id=8994820%40N07&format=json&nojsoncallback=1"})
		jest.resetModules()
		jest.clearAllMocks()
	})
	
it("should be called once", () => { 
	let mockAdapter = new MockAdapter(axios);
	mockAdapter.onGet('https://api.flickr.com/services/rest/?method=flickr.photosets.getPhotos&api_key=0c3f8d32a28de8434240115b85a28499&photoset_id=72157695078896702&user_id=8994820%40N07&format=json&nojsoncallback=1').reply(200, {
       data: {
         pictures: [{farm: '2', server: '1789', id: '42976064761', secret: '38be574628', title: 'Snoqualmie Falls' }]
       }
     });
     let response = App.mounted();
     setTimeout(() => {
			expect(response.pictures[0]).to.be.equal({farm: '2', server: '1789', id: '42976064761', secret: '38be574628', title: 'Snoqualmie Falls' });
		});
});

	
	
	it('equals url to flickr url', () => {
		expect(cmp.vm.url).toEqual("https://api.flickr.com/services/rest/?method=flickr.photosets.getPhotos&api_key=0c3f8d32a28de8434240115b85a28499&photoset_id=72157695078896702&user_id=8994820%40N07&format=json&nojsoncallback=1")
	})



	it('has the expected html structure', () => {
		expect(cmp.element).toMatchSnapshot()
	})
})
