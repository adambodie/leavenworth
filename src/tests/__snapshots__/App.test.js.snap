// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.test.js has the expected html structure 1`] = `
<div
  class="jumbotron"
>
  <!---->
  <h1>
    Photo Gallery
  </h1>
  <div
    class="owl-carousel owl-theme"
  />
</div>
`;
